File: src/etch/bindings/java/support/NullPool.java
package etch.bindings.java.support;

/**
 * A pool which runs the runnable directly.
 */
public class NullPool implements Pool
{
	public void run( PoolRunnable runnable ) throws Exception
	{
		try
		{
			runnable.run();
		}
		catch ( Exception e )
		{
			runnable.exception( e );
		}
	}
}

File: src/etch/bindings/csharp/support/NullPool.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Threading;

namespace Etch.Support
{
    /// <summary>
    /// A pool which runs runnable directly
    /// </summary>
    public class NullPool : Pool
    {
        #region Pool Members

        public void Run( RunDelegate d1, ExceptionDelegate d2 )
        {
            new Thread(
                new ThreadStart( delegate()
                                {
                                    try
                                    {
                                        d1();
                                    }
                                    catch ( Exception e )
                                    {
                                        d2( e );
                                    }
                                } ) ).Start();
        }

        #endregion
    }
}
